src/client/services/cdc/index.js

```diff
@@ -36,40 +35,48 @@ CDC.requestEditKeyValidation(params);
 
 const CDC = {
   initGraphSocket(updateFunction) {
```

Comment:
The socket should only be concerned with sending live position updates to the server.  It should not concern itself with links, permissions, getting graph json, etc.

Reply:
Then should I receive edit links and the graph JSON through a regular API call?

Reply:
Edit links should just be a pattern.  You don't need to ask the server anything.

Getting graph JSON should be an HTTP GET route on the server and you should use `fetch()` to get it.

Reply:
What do you mean by pattern?

Reply:
e.g. /app/layout/edit/<some-id>